File: demo/src/main/java/com/example/demo/entities/StudentFeedback.java
package com.example.demo.entities;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.time.LocalDate;
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "STUDENT_FEEDBACK")
public class StudentFeedback {
    @Id
    @Column(name = "FEEDBACK_ID", nullable = false)
    private Long id;

    @NotNull
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @OnDelete(action = OnDeleteAction.RESTRICT)
    @JoinColumn(name = "STUDENT_ID", nullable = false)
    private Student student;

    @NotNull
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "USER_ID", nullable = false)
    private com.example.demo.entities.User user;

    @NotNull
    @Column(name = "FEEDBACK_DATE", nullable = false)
    private LocalDate feedbackDate;

    @Size(max = 255)
    @NotNull
    @Column(name = "PERFORMANCE_NOTES", nullable = false)
    private String performanceNotes;

    @Size(max = 255)
    @Column(name = "BEHAVIOR_NOTES")
    private String behaviorNotes;

    @Size(max = 255)
    @Column(name = "RECOMMENDATIONS")
    private String recommendations;


}

File: Front/ntg_admin/src/app/Models/student-feedback.ts
export interface StudentFeedbackModel {
  id: number;
  studentId: number;
  studentFirstName: string;
  studentLastName: string;
  userId: number;
  userFirstName: string;
  userLastName: string;
  feedbackDate: string;
  performanceNotes: string;
  behaviorNotes: string | null;
  recommendations: string | null;
}

export interface CreateStudentFeedbackRequest {
  studentId: number;
  userId: number;
  feedbackDate: string;
  performanceNotes: string;
  behaviorNotes?: string;
  recommendations?: string;
}